### PR TITLE
Infrastructure - Add and call vm-provision-fixup.sh

### PR DESCRIPTION
**Pull request by jwhonce**
* Fix issues with base image before proceeding
*  remove 'options single-request-reopen' from resolv.conf, breaks dns
   server fail-over

### CODE REVIEW

**Comment by jwhonce** [comment]
@danmcp @smarterclayton This change corrects DNS resolution when failing over from ipv4 to ipv6. With this change, no code changes are required in origin or it's deps.

@danmcp I successfully ran your box builder script after making this change.

File: hack/vm-provision-fixup.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+
+# patch incompatible with fail-over DNS setup
+SCRIPT='/etc/NetworkManager/dispatcher.d/fix-slow-dns'
+if [[ -f "${SCRIPT}" ]]; then
+    echo "Remove ${SCRIPT}..."
```
Comment (danmcp) [inline hack/vm-provision-fixup.sh]: Nit: Shouldn't this be "Removing ..."

File: hack/vm-provision-full.sh [hunk 2]
```diff
@@ -3,14 +3,21 @@ set -euo pipefail
 IFS=$'\n\t'
 USERNAME="${1:-vagrant}"
 
+
```
Comment (danmcp) [inline hack/vm-provision-full.sh]: Nit: Why another newline?

**Comment by jwhonce** [comment]
@danmcp nits unpicked. :-)

**Comment by danmcp** [comment]
[merge]

**Comment by smarterclayton** [comment]
We need better checks and documentation around this - I think other people are hitting it and have no way of knowing they have to fix something.  Please open an issue so we can get that doc and checks.

**Pull request merged**